unison-cli/src/Unison/Codebase/Transcript/Runner.hs

```diff
@@ -204,12 +205,23 @@ run isTest verbosity dir codebase runtime sbRuntime nRuntime ucmVersion baseURL
           -- We shorten the terminal width, because "Transcript" manages a 2-space indent for output lines.
           Pretty.toPlain (terminalWidth - 2) line
 
-      maybeDieWithMsg :: String -> IO ()
+      maybeDieWithMsg :: Pretty.Pretty Pretty.ColorText -> IO ()
       maybeDieWithMsg msg = do
-        errOk <- readIORef allowErrors
-        if errOk
-          then writeIORef hasErrors True
-          else dieWithMsg msg
+        liftIO $ writeIORef hasErrors True
+        liftIO (liftA2 (,) (readIORef allowErrors) (readIORef expectFailure)) >>= \case
+          (False, False) -> liftIO . dieWithMsg $ Pretty.toPlain terminalWidth msg
+          (True, True) ->
+            transcriptFailure
+              out
+              ( Text.pack . Pretty.toPlain terminalWidth $
+                  "The stanza above previously had an incorrect successful result, but now fails with"
+                    <> "\n"
+                    <> Pretty.border 2 msg
+                    <> "\n"
+                    <> "if this is the expected result, remove `:bug`, otherwise remove `:error`."
```

Comment:
```suggestion
                    <> "If the stanza should fail, remove `:bug`; otherwise remove `:error`."
```

Reply:
Oh, this is the tail of the same message – I missed that.

So, I don’t think my original or the new one are correct here. Removing `:error` is probably not what we want to do. I think what I was trying to capture here is something like “this is now erroring, which means we should remove `:bug` … but is the error the error that we’re actually expecting, or is it still the wrong error message?”

Reply:
I opened a new thread below that doesn't split the message up into two conversations.